### PR TITLE
Use yaml for PR build

### PR DESCRIPTION
**Pull request by LTA-Thinking**
## Description
Sets up the PR build to run off a yaml file.

### CODE REVIEW

File: build/jobs/add-aad-test-environment.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+jobs:  
+- job: setup
+  pool:
+    vmImage: 'vs2017-win2016'
```
Comment (brendankowitz) [inline build/jobs/add-aad-test-environment.yml]: Lets use `windows-latest` https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops

File: build/jobs/cleanup.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+jobs:
+- job: DeleteResourceGroup
+  displayName: 'Delete resource group'
+  pool:
+    vmImage: 'vs2017-win2016'
```
Comment (brendankowitz) [inline build/jobs/cleanup.yml]: Can this be a pipeline-wide variable?

File: build/jobs/cleanup.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  steps:
+  - task: AzurePowerShell@4
+    displayName: 'Delete resource group'
+    inputs:
+      azureSubscription: 'Microsoft Health Open Source Subscription'
```
Comment (brendankowitz) [inline build/jobs/cleanup.yml]: Also variable

File: build/jobs/run-tests.yml [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  - task: DownloadBuildArtifacts@0
+    inputs:
+      buildType: 'current'
+      downloadType: 'single'
+      downloadPath: '$(System.ArtifactsDirectory)'
+      artifactName: 'IntegrationTests'
+
+  - task: UseDotNet@2
+    inputs:
+      version: '3.1.100'
```
Comment (brendankowitz) [inline build/jobs/run-tests.yml]: Should be like this one https://github.com/microsoft/fhir-server/blob/master/build/build.yml#L9
`useGlobalJson: true` I think there should be an input for where to find it as an artifact

File: build/jobs/cleanup.yml (hunk 2 above)
Comment (LTA-Thinking) [reply build/jobs/cleanup.yml]: Changed

File: build/jobs/add-aad-test-environment.yml (hunk 1 above)
Comment (LTA-Thinking) [reply build/jobs/add-aad-test-environment.yml]: Changed

File: build/jobs/cleanup.yml (hunk 3 above)
Comment (LTA-Thinking) [reply build/jobs/cleanup.yml]: Refactored

File: build/jobs/run-tests.yml (hunk 4 above)
Comment (LTA-Thinking) [reply build/jobs/run-tests.yml]: Changed

File: build/jobs/cleanup.yml (hunk 3 above)
Comment (brendankowitz) [reply build/jobs/cleanup.yml:10]: Did this end up working as ${{ConnectedServiceName}}?

File: build/pr-pipeline.yml [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+trigger: none
+
+variables:
+- template: build-variables.yml
+- template: pr-variables.yml
+
+stages:
+- stage: BuildUnitTests
+  displayName: 'Build and run unit tests'
+  dependsOn: []
```
Comment (brendankowitz) [inline build/pr-pipeline.yml:11]: Is this used?

Comment (LTA-Thinking) [reply build/pr-pipeline.yml]: Nope, good catch

File: build/jobs/cleanup.yml (hunk 3 above)
Comment (LTA-Thinking) [reply build/jobs/cleanup.yml]: Changed it everywhere except where you had originally commented...

File: build/pr-pipeline.yml [hunk 6]
```diff
@@ -0,0 +1,195 @@
+
```
Comment (brendankowitz) [inline build/pr-pipeline.yml:1]: Can we add the build name back in?
```
# DESCRIPTION: 	
# Builds, tests and packages the solution for all PR requests.	

name: pr$(system.pullRequest.pullRequestNumber)-$(Date:yyyyMMdd)$(Rev:-r)
```

File: build/pr-pipeline.yml [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+- stage: provisionEnvironment
+  displayName: Provision Environment
+  dependsOn: []
+  jobs:
+  - job: provision
+    steps:
+    - task: AzurePowerShell@4
+      displayName: Provision Resource Group
+      inputs:
+        azureSubscription: 'Microsoft Health Open Source Subscription'
```
Comment (brendankowitz) [inline build/pr-pipeline.yml:39]: One left! `$(ConnectedServiceName)`

Comment (LTA-Thinking) [reply build/pr-pipeline.yml]: I finally did what I should have done in the first place, a global search. This is the last one for certain!

File: build/pr-pipeline.yml (hunk 6 above)
Comment (LTA-Thinking) [reply build/pr-pipeline.yml:3]: Added

**Pull request merged**